aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/LastSyncMetadata.java

```diff
@@ -53,9 +55,11 @@ private LastSyncMetadata(String id, String modelClassName, Long lastSyncTime) {
      * @param lastSyncTime Last time it was synced
      * @return {@link LastSyncMetadata} for the model class
      */
-    static <T extends Model> LastSyncMetadata lastSyncedAt(@NonNull Class<T> modelClass, long lastSyncTime) {
+    static <T extends Model> LastSyncMetadata lastSyncedAt(@NonNull Class<T> modelClass,
+                                                           @Nullable long lastSyncTime,
+                                                           @NonNull SyncType syncType) {
```

Comment:
To cut down on the number of arguments, and since there is already a `create(...)`, how about we replace `lastSyncedAt` with:

```java
static <T extends Model> LastSyncMetadata baseSyncedAt(@NonNull Class<T> modelClass, long lastBaseSyncTime) {
    ....
}

static <T extends Model> LastSyncMetadata deltaSyncedAt(@NonNull Class<T> modelClass, long lastDeltaSyncTime) {
    ....
}

static <T extends Model> LastSyncMetadata neverSynced(@NonNull Class<T> modelClass) {
    ...
}
```